clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -745,6 +751,46 @@ void StreamChecker::evalFreadFwrite(const FnDescription *Desc,
     C.addTransition(StateFailed);
 }
 
+void StreamChecker::evalFgetcFputc(const FnDescription *Desc,
```

Comment:
I do not know if there are enough similarities between `fputc` and `fgetc` to put these in a single function (return value of `fgetc` is differently constructed and different errors can happen including FEOF). The function should now be called only `evalFputc` because there is no "IsRead" argument, if later this is reused for `fgetc` it can be renamed.